crt_portal/cts_forms/models.py

```diff
@@ -545,6 +560,80 @@ class Report(models.Model):
     class Meta:
         indexes = [GinIndex(fields=['violation_summary_search_vector'])]
 
+    def redact(self):
+        """Removes all information from the report besides its ID"""
+
+        if self.litigation_hold:
+            logger.error(f'Attempted to redact report {self.public_id} while on litigation hold.')
+            return
+
+        self.disposed = True
+
+        if not settings.REDACT_REPORTS:
+            self.save()
+            return
+
+        ignore = [
+            'id',
+            'public_id',
+            'modified_date',
+            'retention_schedule',
+            'batched_for_disposal',
+            'disposed',
+            'email_report_count',
```

Comment:
Can we retain the email_report_count field? Maybe it's abstract enough but you could potentially tie it to the email report count of an open report

Reply:
this is a materialized view - it's not actually stored anywhere, it's the result of a SQL query that's continuously rerun

So if we want to retain this, we need to rebuild how that works to not include a materialized view